kore/src/Kore/Step/Step.hs

```diff
@@ -438,7 +448,64 @@ recoveryFunctionLikeResults
     => Pattern (Target variable)
     -> unifier (Results variable)
     -> unifier (Results variable)
-recoveryFunctionLikeResults _ = id
+recoveryFunctionLikeResults initial unifier = do
+    results <- Results.results <$> unifier
+    let appliedRules = Result.appliedRule <$> results
+    case checkFunctionLike appliedRules (Conditional.term initial) of
+        Right _  -> unifier
+        Left err -> moreChecksAfterError appliedRules err
+  where
+    moreChecksAfterError appliedRules err = do
+        let
+            appliedRule = case Seq.length appliedRules of
+                1 -> Seq.index appliedRules 0
```

Comment:
We can match on `appliedRules` using pattern synonyms instead of applying a partial function:

```suggestion
                rule Seq.:< Seq.EmptyL -> rule
```